app/stores/game-store.js

```diff
@@ -148,6 +149,55 @@ async function fetch_keys (type, page) {
   cache_owned_games()
 }
 
+async function fetch_search_page (query, page, game_ids, games) {
+  if (typeof page === 'undefined') {
+    page = 1
+  }
+
+  if (typeof game_ids === 'undefined') {
+    game_ids = []
+  }
+
+  if (typeof games === 'undefined') {
+    games = {}
+  }
+
+  log(opts, `fetching page ${page} of search '${query}'`)
+
+  let user = CredentialsStore.get_current_user()
+
+  let res = await user.search(query, page)
+  let total_items = res.total_items || res.per_page // FIXME: remove || res.per_page
+  let fetched = res.per_page * page
+  game_ids = game_ids.concat(_.pluck(res.games, 'id'))
+  for (let game of res.games) {
+    games[game.id] = game
+  }
+
+  await db.save_games(res.games)
+  AppActions.search_fetched(query, game_ids, games)
+
+  if (fetched < total_items) {
+    await fetch_search_page(query, page + 1, games)
```

Comment:
Hmm, we probably shouldn't fetch all search pages in one go - as opposed to collections which we want to have available offline